advanced/matrix/matrix.py

```diff
@@ -0,0 +1,178 @@
+        self._check_tuple()
+        self._check_even()
+
+    @property
+    def tuples(self):
+        """Return the matrix."""
+        return self.__matrix
+
+    @classmethod
+    def unity(cls, size):
+        """Create a unity tuple.
+
+        Args:
+            size (int): the number of rows and columns for new matrix.
+
+        Returns:
+            tuple: A New matrix where main diagonal is 1, the rest 0.
+        """
+        u_matrix = tuple(tuple(1 if col == row else 0 for col in range(size))
+                         for row in range(size))
+
+        return cls(u_matrix)
+
+    @classmethod
+    def ones(cls, size):
+        """Create a new matrix in the given size filled with 1.
+
+        Args:
+            size: The size of the new matrix.
+
+        Returns:
+            tuple: Tuple of tuples filled with ones.
+        """
+        ones_mat = (((1,) * size,) * size)
+        return cls(ones_mat)
+
+    def _multiplication(self, other_matrix):
+        """Calculate the multiplication of 2 matrices.
+
+            Args:
+                other_matrix (tuple): A matrix to multiply with.
+
+            Returns:
+                tuple: Result matrix of the multiplication between 2 matrices.
+        """
+        result = tuple(tuple(sum(n_a * n_b for n_a, n_b in zip(a_row, b_col))
+                       for b_col in zip(*other_matrix))
+                       for a_row in self)
+        return self.__class__(result)
+
+    def _multiply_by_scalar(self, scalar):
+        """Return new matrix multiplied by scalar.
+
+        Args:
+            scalar (int/float): a number to multiply matrix with.
+
+        Returns:
+            Matrix : The new matrix multiplied by scalar.
+
+        """
+        return self.__class__(tuple(tuple(num * scalar for num in each_tuple)
+                              for each_tuple in self))
+
+    def __mul__(self, other):
+        """Multiply matrix by scalar or another matrix.
+
+        Args:
+            other (float/int/Matrix): Scalar or another matrix.
+
+        Returns:
+            Matrix: The result matrix.
+        """
+        try:
+            self._check_valid_matrix(other)
+            return self._multiplication(other.tuples)
+
+        except TypeError:
+            if isinstance(other, int) or isinstance(other, float):
+                return self._multiply_by_scalar(other)
+
+            else:
+                raise TypeError("Can only multiply numbers and matrices.")
+
+    def _check_valid_matrix(self, other):
+        """Check that 'other' is a matrix of the same size as self."""
+        if isinstance(other, Matrix):
+            if len(other.tuples) != len(self.tuples):
+                raise ValueError("Cannot operate on different sized matrices.")
+
+        else:
+            raise TypeError("Cannot operate on different sized matrices.")
+
+    def __rmul__(self, other):
+        """Return the regular multiplication when in reversed order."""
+        return self * other
```

Comment:
If other is matrix this is not correct.
Can you tell me why even so, this answer is still correct?